### PR TITLE
Remove OJ Dependency

### LINKED ISSUE #34: Drop Oj as a dependency
Is there any reason that this gem must have `Oj` as a dependency?

I know it speeds up JSON generation, but if it's not strictly necessary, it should be removed. I work on a project that can't have `Oj` installed, due to some specific JSON interactions between other gems and services. What I had to do was to manually set `MultiJson` to ignore `Oj` and use the default JSON adapter, but that's not ideal.

Would you consider dropping `Oj` as a dependency here?

### LINKED ISSUE #44: Drop Oj as a Dependency
Is there a reason you have Oj as a required dependency?  The whole point of using multi_json as a gem is that it allows users of your gem to choose the JSON coder they want to use.  By requiring Oj as a dependency, you remove that choice and lock them into using Oj.

Personally I would like to be able to use the JSON coder that ships with Ruby.  Oj by default serializes floating point numbers differently from the Ruby JSON coder.

```
irb(main):099:0> JSON.parse("{\"id\":1,\"rating\":7.2727272727272721231231235}")['rating'].class
=> Float
irb(main):100:0> Oj.load("{\"id\":1,\"rating\":7.2727272727272721231231235}")['rating'].class
=> BigDecimal
```

This means that even though I am serializing something as a number, Olive Branch is parsing it as a BigDecimal and then re-serializing it as a string.

I was able to get around this by changing the default options in Oj: 

```ruby
Oj.default_options = { bigdecimal_load: :float }
```

but I could easily see someone deciding to use this gem without knowing this could be an issue.

### PR DESCRIPTION
**Pull request by sdrioux**
Closes #44
Closes #34 

This gem uses multi_json under the hood to choose the fastest JSON coder available in the application.  By requiring Oj as a dependency, we remove that choice and require the application to deserialize and re-serialize each response using Oj, which may result in a different JSON response.  For example, Oj sometimes de-serializes floating numbers as BigDecimals instead of Floats.  BigDecimals then get serialized as strings instead of the original floating number.  This is a different implementation from the JSON coder that ships with Ruby.

As this is is a breaking change from 3.0.0, this should probably be considered a new major version.  I noticed there wasn't a CHANGELOG.md file set up yet.  I added one and set it up with the most recent version.

I also added the appraisal configuration to start testing this gem against the latest version of Rails.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,16 +6,27 @@ rvm:
...
-    - rvm: 2.6.1
+      gemfile: gemfiles/rails_6.1.gemfile
+    - rvm: 2.3
+      gemfile: gemfiles/rails_6.0.gemfile
+    - rvm: 2.3
+      gemfile: gemfiles/rails_6.1.gemfile
+    - rvm: 2.4
       gemfile: gemfiles/rails_6.0.gemfile
+    - rvm: 2.4
+      gemfile: gemfiles/rails_6.1.gemfile
```
Comment (sdrioux) [inline .travis.yml:32]: This last part was set up incorrectly and was being ignored by Travis.  I replaced it with an exclude list to ignore combinations that are not supported by Rails.

**Review by efatsi — APPROVED** [review]
This is awesome. Thanks @sdrioux!

Looks like Ruby 2.7.3 doesn't play well with Rails 4.2 (bundler incompatibility). Up to you if you want to tackle that for the Travis ✅, or just add that pair to the exclusions list (if that's how that works).

**Comment by sdrioux** [comment]
Oh interesting.  I’ll add it as an exclusion - really doubt anyone is using Ruby 2.7 and Rails 4.

**Comment by efatsi** [comment]
Thanks again @sdrioux! I'm releasing version 4.0.0 shortly.

**Pull request merged**